Integrations/integration-Cymon.yml

```diff
@@ -0,0 +1,543 @@
+
+        return md
+
+    def create_context_domain_command(domain, reports):
+
+        cymon_domain_context_activities = []
+        description = 'Reported suspicious activities: '
+
+        for report in reports:
+            cymon_domain_context_activities.append({
+                'Title': report.get('Title'),
+                'Tags': report.get('Tags'),
+                'Time': report.get('Timestamp'),
+                'Hostname': report.get('Hostname'),
+                'IP': report.get('IP')
+            })
+
+            description += '{}, '.format(report.get('Title'))
+
+        description = description[:-2]
+
+        context = {
+            outputPaths['domain']: {
+                'Name': domain,
+                'Malicious': {
+                    'Vendor': 'Cymon',
+                    'Description': description
+                }
+            },
+            'Cymon': {
+                'Domain': {
+                    'Activities': cymon_domain_context_activities
+                }
+            }
+        }
+
+        return context
+
+    def get_domain_report_command():
+        args = demisto.args()
+
+        domain = args.get('domain')
+        full_response = args.get('fullResponse') == 'true'
+
+        domain_report = get_domain_report(domain)
+        domain_full_report = []
+
+        if full_response:
+            domain_full_report = get_domain_full_report(domain)
+
+        if not domain_report.get('reports') or not domain_report.get('total'):
+            return "Domain " + domain + " is not in Cymons's dataset"
+
+        mark_down = create_domain_command_markdown(domain, domain_report.get('total'), domain_report.get('reports'), domain_full_report)
+        context = create_context_domain_command(domain, domain_report.get('reports'))
+
+        return {
+                'Type': entryTypes['note'],
+                'Contents': domain_report,
+                'ContentsFormat': formats['json'],
+                'HumanReadable': mark_down,
+                'EntryContext': context
+            }
+
+    ''' IP COMMAND '''
+    def get_ip_events_sources(ip):
+        url = '{}/{}/{}'.format(SERVER_URL_V1, 'ip', ip)
+        response = http_request('GET', url, DEFAULT_HEADERS)
+
+        return response.get('sources', None)
+
+    def get_ip_events(ip):
+        url = '{}/{}/{}/{}?limit={}'.format(SERVER_URL_V1, 'ip', ip, 'events', 100)
+        events = {}
+
+        next_link = url
+
+        while next_link is not None:
+            response = http_request('GET', next_link, DEFAULT_HEADERS)
+
+            for event in response.get('results', []):
+                tag = event.get('tag')
+                date = datetime.datetime.strptime(event.get('updated'), '%Y-%m-%dT%H:%M:%SZ')
+
+                if (tag in events and events[tag] < date) or tag not in events:
```

Comment:
@MarkShleifer make sure that the comparison is valid (also in the case we don't have a `date`

Reply:
Done